src/assets/thirdPartyServices/snet/minecraftizing_service/index.js

```diff
@@ -0,0 +1,195 @@
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  submitAction() {
+    const { methodName, model_name, dataset, input_image } = this.state;
+    const methodDescriptor = MinecraftizingService[methodName];
+    const request = new methodDescriptor.requestType();
+
+    request.setNetworkName(model_name);
+    request.setDataset(dataset);
+    request.setInputImage(input_image);
+
+    const props = {
+      request,
+      onEnd: ({ message }) => {
+        this.setState({
+          ...initialUserInput,
+          response: { status: message.getStatus(), data: message.getOutput(), source: input_image},
+        });
+      },
+    };
+
+    this.props.serviceClient.unary(methodDescriptor, props);
+  }
+
+  getImageData(data)
+  {
+    this.setState({input_image: data})
+  }
+
+  renderForm() {
+    const serviceNameOptions = ["Select a method", ...this.props.serviceClient.getMethodNames(MinecraftizingService)];
+    const modelNames = ["Select model", "UGATIT", "cycle_gan"];
+    const datasetNames = ["Select dataset", "minecraft_landscapes"];
+
+    return (
+      <React.Fragment>
+        <div className="row">
+          <div className="col-md-3 col-lg-3" style={{ padding: "10px", fontSize: "13px", marginLeft: "10px" }}>
+            Method Name:{" "}
+          </div>
+          <div className="col-md-3 col-lg-3">
+            <MethodNamesDropDown
+                list={serviceNameOptions}
+                value={this.state.methodName}
+                onChange={this.handleFormUpdate}
+            />
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-md-3 col-lg-3" style={{ padding: "10px", fontSize: "13px", marginLeft: "10px" }}>
+            Detector Name:{" "}
+          </div>
+          <div className="col-md-3 col-lg-3">
+            <RandomDropDown
+                name={"model_name"}
+                list={modelNames}
+                value={this.state.model_name}
+                onChange={this.handleFormUpdate}
+            />
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-md-3 col-lg-3" style={{ padding: "10px", fontSize: "13px", marginLeft: "10px" }}>
+            Detector Name:{" "}
+          </div>
+          <div className="col-md-3 col-lg-3">
+            <RandomDropDown
+                name={"dataset"}
+                list={datasetNames}
+                value={this.state.dataset}
+                onChange={this.handleFormUpdate}
+            />
+          </div>
+        </div>
+        <div className="row" align="center">
+          <SNETImageUpload imageName={""} imageDataFunc={this.getImageData} instantUrlFetch={true} allowURL={true} />
+        </div>
+        <div className="row">
+          <div className="col-md-3 col-lg-3" style={{ padding: "10px", fontSize: "13px", marginLeft: "10px" }}>
+            About:{" "}
+          </div>
+          <div className="col-xs-3 col-xs-2">
```

Comment:
this div is repeating. extract it as a separate component. The data could be kept in an array and the same could be iterated using the map function.